lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -150,11 +208,17 @@ def validate(self):
         validation = super(LmsBlockMixin, self).validate()
         has_invalid_user_partitions = False
         has_invalid_groups = False
+
         for user_partition_id, group_ids in self.group_access.iteritems():
             try:
                 user_partition = self._get_user_partition(user_partition_id)
             except NoSuchUserPartitionError:
                 has_invalid_user_partitions = True
+            except AttributeError:
+                try:
+                    user_partition = self.user_partitions[user_partition_id]
```

Comment:
That might be it.  I've tried setting up a runtime with both the basic TestRuntime and with 
```
self.runtime = MagicMock(course_id=self.course.id)
```
which I've found elsewhere in the platform.  Both of those give me the same AttributeError so I was wondering if you had any other suggestions I could try.